tests/ttnn/unit_tests/operations/test_moreh_logsoftmax.py

```diff
@@ -19,10 +19,127 @@
 )
 
 
+def get_torch_dtype(dtype):
+    if dtype == ttnn.int32:
+        return torch.int32
+    elif dtype == ttnn.float32:
+        return torch.float32
+    else:
+        return torch.bfloat16
+
+
+def run_moreh_logsoftmax_test(
+    shape,
+    dim,
+    ttnn_dtype,
+    layout,
+    device,
+    rtol,
+    atol,
+    use_randint,
+    multiple_of_32,
+    optional_output_tensor=False,
+    strategy=None,
+    compute_kernel_options=None,
+):
+    if ttnn_dtype == ttnn.bfloat8_b:
+        pytest.skip(f"bfloat8_b is not supported")
+    torch_dtype = get_torch_dtype(ttnn_dtype)
+    if use_randint == True:
+        torch_input = torch.randint(low=0, high=4, size=shape).to(torch_dtype) + 100
+    else:
+        torch_input = torch.rand(size=shape, dtype=torch_dtype) + 100
+    ttnn_input = ttnn.from_torch(torch_input, dtype=ttnn_dtype, layout=layout, device=device)
+
+    torch_output = F.log_softmax(torch_input, dim)
+
+    if optional_output_tensor == True:
+        optional_output = ttnn.from_torch(torch_input, dtype=ttnn_dtype, layout=layout, device=device)
+        ttnn_output = ttnn.operations.moreh.logsoftmax(ttnn_input, dim, output_tensor=optional_output)
+    elif compute_kernel_options is not None:
+        compute_kernel_config = get_compute_kernel_options(compute_kernel_options)
+        if strategy is None:
+            ttnn_output = ttnn.operations.moreh.logsoftmax(ttnn_input, dim, compute_kernel_config=compute_kernel_config)
+        else:
+            ttnn_output = ttnn.operations.moreh.logsoftmax(
+                ttnn_input, dim, compute_kernel_config=compute_kernel_config, strategy=strategy
+            )
+    else:
+        ttnn_output = ttnn.operations.moreh.logsoftmax(ttnn_input, dim, strategy=strategy)
+
+    if multiple_of_32 == False:
+        ttnn_output = ttnn_output.cpu().to(ttnn.ROW_MAJOR_LAYOUT).unpad_from_tile(shape)
+
+    assert list(ttnn_output.shape.with_tile_padding()) == list(torch_output.shape)
+
+    ttnn_output = ttnn.to_torch(ttnn_output)
```

Comment:
Because you are testing the same ttnn and torch dtype (fp32 vs fp32, bfloat16 vs bfloat16) so this is correct. But if you test with different dtype between npu and cpu then you have to do
```python
ttnn_output = ttnn.to_torch(ttnn_output).to(cpu_dtype)
```

Reply:
Oh I see, I will fix it right away

Reply:
Resolved